src/sentry/search/events/builder.py

```diff
@@ -112,6 +112,7 @@ def __init__(
         # This allows queries to be resolved without adding time constraints. Currently this is just
         # used to allow metric alerts to be built and validated before creation in snuba.
         skip_time_conditions: bool = False,
+        parser_config_overrides: Optional[Mapping[str, Any]] = None,
```

Comment:
Dang its too bad we can't use the `SearchConfig` here somehow instead of `Any`

Reply:
Yeah it'd be pretty nice... I think that unfortunately it might result in us overwriting a bunch of values. I'll spend a few minutes thinking about it just in case I can think of something better